README.md

```diff
@@ -86,101 +143,121 @@ The only required section is `site_info`
+
+#### Themes and plugins
+
+The Cloud Foundry platform builds apps with ephemeral local storage. This means any changes made to local files on your app will get deleted whenever you `push` or `restage` the app. Make sure your plugins and themes remain installed by installing them through the `plugins` and `themes` sections of the `setup.json` file.
+
+For plugins or themes you'd normally be able to install from the admin interface, you can list them by name and the version that you want installed. For anything not available through WordPress directly, provide a URL to a ZIP file. For example, if your site's theme is one you've developed custom, you can host it on GitHub and use the "Download ZIP" link to install it.
+
+```json
+"plugins": [
     {
-      "name": "amazon-web-services",
-      "version": "0.3.4"
+      "url": "https://github.com/humanmade/S3-Uploads/archive/v1.1.0.zip"
     },
     {
-      "name": "amazon-s3-and-cloudfront"
+      "name": "akismet",
+      "version": "4.0"
     },
     {
-      "url": "https://example.org/my-great-plugin-0.1.zip"
+      "name": "hello-dolly"
     }
-  ],
-  "themes": [
+],
+"themes": [
     {
-      "name": "create",
-      "version": "1.3"
+        "name": "create",
+        "version": "1.3"
     }
   ]
-}
 ```
 
-### How It Works
+In the sample `setup.json` file you see three plugins installed by default. One from GitHub and two from WordPress, one of which is pinned to a specfic version. As with WordPress Core, if you pin the version, make sure to watch for and install updates that contain security fixes.
 
-When you push the application here's what happens.
+#### Running WP-CLI commands
 
-1. The local bits are pushed to your target.  This is small, five files around 25k. It includes the changes we made and a build pack extension for Wordpress.
-1. The server downloads the [PHP Build Pack] and runs it.  This installs HTTPD and PHP.
-1. The build pack sees the extension that we pushed and runs it.  The extension installs [wpcli](http://wp-cli.org/), and then reads your `setup.json` to install WordPress, and any plugins or themes you have defined. It then copies the rest of the files that we pushed and replaces the default Wordpress files with them.  In this case, it's just the `wp-config.php` file.
-1. At this point, the build pack is done and CF runs our droplet.
+We recommend using Cloud Foundry's "tasks" to run `wp-cli` commands. To do this, make sure to give the full path for both PHP and the `wp-cli.phar` file and specify the WordPress path relative to the `app` directory. Here's how you'd run `wp core version` on your cloud.gov container:
 
-### Changes
+```bash
+cf run-task APP_NAME "php/bin/php htdocs/wp-cli.phar core version --path='htdocs/'"
```

Comment:
Why not `cf ssh APP_NAME --command "php/bin/php htdocs/wp-cli.phar core version --path='htdocs/'"`? Then you can skip the asynchronous "see the results in the logs" malarkey.

Reply:
I didn't have a ton of luck with `cf ssh` but I'll give it another shot because tailing the logs is kind of a pain.

Reply:
Ah, yeah, I remember what it was now. If you use cf ssh you have to give the full path, including `/home/vcap/app` for PHP and the `htdocs` directory. So the command ends up being a bit longer `cf ssh APP_NAME --command "/home/vcap/app/php/bin/php /home/vcap/app/htdocs/wp-cli.phar core version --path='/home/vcap/app/htdocs/'"`

And some commands exit with `Fatal error: Call to undefined function mysql_connect()`, I think because the `cf ssh` doesn't bind the RDS? Maybe?

Reply:
It should bind properly. You might have to do some of the stuff described on https://docs.cloudfoundry.org/devguide/deploy-apps/ssh-apps.html#ssh-env to get the environment for SSH match the environment for the app. I'm sure someone on the Platform Squad can help you make it a short one-liner.